crypto/signing/mcl/multisig/bls.go

```diff
@@ -133,20 +117,37 @@ func (bms *BlsMultiSigner) VerifyAggregatedSig(
 }
 
 func preparePublicKeys(
-	pubKeysPoints []crypto.Point,
+	pubKeys []crypto.PublicKey,
 	hasher hashing.Hasher,
 	suite crypto.Suite,
 ) ([]bls.PublicKey, error) {
-	prepPubKeysPoints := make([]bls.PublicKey, len(pubKeysPoints))
+	prepPubKeysPoints := make([]bls.PublicKey, len(pubKeys))
+
+	concatPKs, err := concatPubKeys(pubKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	pubKeysPoints := make([]crypto.Point, len(pubKeys))
+	for i, pubKey := range pubKeys {
+		if check.IfNil(pubKey) {
```

Comment:
Why not make this check in the next for and get the point there?

Reply:
right, done